client/components/orders/CurrentOrder.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory, Link } from 'react-router-dom';
+import axios from 'axios';
+
+import { CurrentOrderCard, CurrentOrderHeader, CurrentOrderPage, CurrentOrderForm, Button } from '../../styles/CurrentOrderStyles';
+
+const CurrentOrder = () => {
+
+  const [order, setOrder] = useState({})
+  const [drinks, setDrinks] = useState([])
+  const [loading, setLoading] = useState(true)
+  const [subtotal, setSubTotal] = useState(0)
+
+  useEffect(async() => {
+    try {
+        const token = window.localStorage.getItem('token')
+        const {data: order} = await axios.get(`/api/user/order/current`, 
+        { headers: { authorization: token } }
+        )
+        setOrder(order)
+        setLoading(false)
+        console.log(order)
+
+        const {data} = await axios.get(`/api/user/order/${order.id}`, 
+            { headers: { authorization: token } }
+            )
+            const drinks = data.orderDrinks 
+            setDrinks(drinks)
+            console.log(drinks)
+      }
+      catch (ex) {
+        console.log(ex)
+      } 
+  }, [])
+
+  useEffect(() => {
+    if (drinks.length) {
+        const prices = []
+        drinks.map(drink => {
+            prices.push(drink.price)
+        })
+        const subtotal = prices.reduce((acc, cum) => acc + cum)
+
+        setSubTotal(subtotal)
+    }
+  })
+
+  const increment = (id, subtotal) => {
+    let newSubtotal; 
+    const newDrinks = drinks.map((drink) => {
+      if (drink.drink.id === id) {
+        drink.quantity++
+        newSubtotal  = subtotal + drink.drink.price
+      }
+      return drink
+    })
+    setDrinks(newDrinks)
+    setSubTotal(newSubtotal)
+  }
+
+   const decrement = async(id, subtotal) => {
+    let newSubtotal;
+    const newDrinks = drinks.map((drink) => {
+      if (drink.drink.id === id) {
+        if (drink.quantity > 0) {
+          drink.quantity--
+          newSubtotal  = subtotal - drink.drink.price
+        }
+      }
+      return drink
+    }).filter(drink => drink.quantity !== 0)
+    // update drink here with put route?
+    setDrinks(newDrinks)
+    setSubTotal(newSubtotal)
+  }
```

Comment:
I think we'll also want these to send PUT requests to the server to modify the order in our database, right?

Reply:
definitely, I believe @arjanmitra is working on these